gslib/utils/boto_util.py

```diff
@@ -294,6 +295,14 @@ def GetNewHttp(http_class=httplib2.Http, **kwargs):
   http = http_class(proxy_info=proxy_info, **kwargs)
   http.disable_ssl_certificate_validation = (not config.getbool(
       'Boto', 'https_validate_certificates'))
+
+  global_context_config = context_config.get_context_config()
+  if global_context_config and global_context_config.use_client_certificate:
+    http.add_certificate(key=global_context_config.client_cert_path,
```

Comment:
Can we add a unit test to check that add certificate gets called if context config is present?

Reply:
Added a new test file, same reasoning as above